### ISSUE TITLE
Angular 4 fix. ERROR in Error encountered resolving symbol values statically...

### ISSUE THREAD

**Issue by pagner** [l-lin/angular-datatables#1015]
I just wanted to share my angular 4 fix with everyone. 

If you are getting the following error: "ERROR in Error encountered resolving symbol values statically...."

Add .forRoot() to your imports for DataTablesModule.

    imports: [
        DataTablesModule.forRoot()
    ]
Hope this helps someone. Took me a bit to find a fix.

**Comment by l-lin** [comment]
I released the version [4.0.0](https://github.com/l-lin/angular-datatables/releases/tag/v4.0.0).
It should be ok now without the `.forRoot()`. Tell me if you still have trouble using with Angular 4.x.x.

**Comment by pagner** [comment]
Hey thanks a lot for your quick fix!

There is another issue I am having when using datatables "the angular way".
When adding and removing rows, the row count is not updated. I have been
using the rerender() option, but if gives undesired results with the table
flashing and jumping even though it does work, it's not smooth.

Thanks for building this directive. It's very nice!

Paul

**Comment by pagner** [comment]
Hi there,
Your fix was working great. I was able to remove the forRoot(). Angular 4
was updated recently and DataTables is once again broken. I've had to
replace the .forRoot() to get my project to compile.

Also, have you been able to address the other problem? I will paste it
below incase you don't have it...

Thanks,
Paul

There is another issue I am having when using datatables "the angular way".
When adding and removing rows, the row count is not updated. I have been
using the rerender() option, but if gives undesired results with the table
flashing and jumping even though it does work, it's not smooth.
